client/state/data-layer/wpcom/user-settings/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, keys, merge, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { decodeEntities } from 'lib/formatting';
+import { http } from 'state/data-layer/wpcom-http/actions';
+import { dispatchRequest } from 'state/data-layer/wpcom-http/utils';
+import { updateUserSettings, clearUnsavedUserSettings } from 'state/user-settings/actions';
+import {
+	USER_SETTINGS_REQUEST,
+	USER_SETTINGS_SAVE,
+} from 'state/action-types';
+
+/*
+ * Decodes entities in those specific user settings properties
+ * that the REST API returns already HTML-encoded
+ */
+function fromApi( apiResponse ) {
+	const decodedValues = {
+		display_name: apiResponse.display_name && decodeEntities( apiResponse.display_name ),
+		description: apiResponse.description && decodeEntities( apiResponse.description ),
+		user_URL: apiResponse.user_URL && decodeEntities( apiResponse.user_URL )
+	};
+
+	return merge( {}, apiResponse, decodedValues );
+}
+
+export const requestUserSettings = ( { dispatch }, action, next ) => {
+	dispatch( http( {
+		apiVersion: '1.1',
+		method: 'GET',
+		path: '/me/settings',
+	}, action ) );
+
+	return next( action );
+};
+
+export const receiveUserSettings = ( { dispatch }, action, next, data ) => {
+	dispatch( updateUserSettings( fromApi( data ) ) );
+};
+
+export const receiveRequestError = noop;
+
+/*
+ * Post settings to WordPress.com API at /me/settings endpoint
+ */
+export function saveUserSettings( { dispatch, getState }, action, next ) {
+	const { settingsOverride } = action;
+	const settings = settingsOverride || getState().userSettings.unsavedSettings;
```

Comment:
> thanks for the explanation. it still seems unclear to me what the intention of this code is. did I correctly infer that in my comment about?

Yes, that's exactly right.

I'm adding a new `getUnsavedUserSettings` selector and using it instead of directly accessing the state.